x/concentrated-liquidity/types/keys.go

```diff
@@ -198,3 +207,18 @@ func KeyUptimeAccumulator(poolId uint64, uptimeIndex uint64) string {
 	uptimeIndexStr := strconv.FormatUint(uptimeIndex, uintBase)
 	return strings.Join([]string{string(UptimeAccumulatorPrefix), poolIdStr, uptimeIndexStr}, "/")
 }
+
+// Helper Functions
+
+func MustGetPoolIdFromShareDenom(denom string) uint64 {
+	parts := strings.Split(denom, "/")
+	numberStr := ""
+	if len(parts) >= 3 {
+		numberStr = parts[2]
+	}
```

Comment:
Can we do a strict length check? So that this panics if the number of parts is not as expected

Reply:
https://github.com/osmosis-labs/osmosis/pull/4882/commits/5cb0a2d99233a3d9c1a32edc3666340b99759ad3